static/elements/chromedash-sample-panel.js

```diff
@@ -0,0 +1,178 @@
+  filter(query, category) {
+    let features = this.features;
+
+    if (!features) {
+      return;
+    }
+
+    if (!query && !category) {
+      if (history && history.replaceState) {
+        history.replaceState('', document.title, location.pathname + location.search);
+      } else {
+        location.hash = '';
+      }
+      this.filtered = features;
+      this._fireEvent('update-length', {length: this.filtered.length});
+
+      return;
+    }
+
+    if (query) {
+      if (history && history.replaceState) {
+        // TODO debounce this 500ms
+        history.replaceState({}, document.title, '/samples#' + query);
+      }
+
+      // Returns operator and version query e.g. ["<=25", "<=", "25"].
+      const operatorMatch = /^([<>=]=?)\s*([0-9]+)/.exec(query);
+      if (operatorMatch) {
+        features = this._filterOnOperation(features, operatorMatch[1], operatorMatch[2]);
+      } else {
+        const regex = new RegExp(query, 'i');
+        features = features.filter((feature) => {
+          return regex.test(feature.name) || regex.test(feature.summary);
+        });
+      }
+    }
+
+    // Further refine list based on selected category in menu.
+    if (category) {
+      const regex = new RegExp(category, 'i');
+      features = features.filter((feature) => {
+        return regex.test(feature.category);
+      });
+    }
+
+    this.filtered = features;
+
+    this._fireEvent('update-length', {length: this.filtered.length});
+  }
+
+  render() {
+    if (!this.filtered) {
+      return html``;
+    }
+    return html`
+      <link rel="stylesheet" href="/static/css/elements/chromedash-sample-panel.css">
+
+      <ul>
+        ${this.filtered.map((feature) => html`
+          <li>
+            <div class="card">
+              <h3 class="feature-name">
+                <a href="${this._computeFeatureLink(feature.id)}">${feature.name}</a>
+                <span class="milestone" ?hidden="${!feature.shipped_milestone}">Chrome <span>${feature.shipped_milestone}</span></span>
+              </h3>
+              <div class="summary">${feature.summary}</div>
+              <div class="sample_links">
+                <div class="demo-links layout horizontal center">
+                  ${feature.sample_links.map((link) => html`
+                    <a href="${link}" target="_blank" class="demo-link">
+                      <iron-icon icon="chromestatus:open-in-browser"></iron-icon> View Demo
+                    </a>
+                    `)}
+                </div>
+                <iron-icon icon="${this._computeIconId(feature.category)}"
+                           title="${feature.category}"></iron-icon>
+              </div>
+            </div>
+          </li>
+          `)}
+      </ul>
+    `;
+  }
+}
+
+customElements.define('chromedash-sample-panel', ChromedashSamplePanel);
+
+// TODO: move this into a behavior. It's duplicated from chromedash-featurelist.html.
```

Comment:
this doesn't need to be a behavior (I guess it means mixin now?) It can just be a function in a shared module.

Reply:
Looking at the code in both places, they are quite different. I guess this is a very old comment from Eric.